server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/daco/DaCoVisitor.java

```diff
@@ -101,13 +100,26 @@ private List<Node> addTreeNode(ParserRuleContext ctx, Function<Locality, Node> n
   }
 
   private Locality constructLocality(ParserRuleContext ctx) {
-    return Locality.builder()
-        .uri(uri)
-        .range(DialectUtils.constructRange(ctx))
-        .build();
+    Location location = context.getExtendedSource().getMainMap().mapLocation(DialectUtils.constructRange(ctx), false);
+    return Locality.builder().uri(context.getExtendedSource().getUri()).range(location.getRange()).build();
   }
 
   public List<SyntaxError> getErrors() {
     return errors;
   }
+
+  private void addReplacementContext(TerminalNode token) {
+    String newText = context.getExtendedSource().getText()
+            .substring(token.getSymbol().getStartIndex(), token.getSymbol().getStopIndex() + 1)
+            .replaceAll("[^ \n]", " ");
+    Range range = DialectUtils.constructRange(token);
+    context.getExtendedSource().replace(range, newText);
+  }
+  private void addReplacementContext(ParserRuleContext ctx) {
+    String newText = context.getExtendedSource().getText()
```

Comment:
Should we extract this to some utility? I see similar use in [CICSTranslatorVisitor.java](https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/1509/files#diff-f6aab91e7cabc8531a9d4fd397d1c0783ad09d988dbcbebd32561e1e7d6f85ae).

something like, `addReplacementContext(TerminalNode token, String replacementText )`

Reply:
These methods are not wholly the same :-(